lib/str-format.c

```diff
@@ -339,6 +339,29 @@ scan_expect_char(const gchar **buf, gint *left, gchar value)
 }
 
 gboolean
+scan_expect_str(const gchar **buf, gint *left, const gchar *value)
```

Comment:
Because all of our `scan_except` functions has another effect: they update `buf` and `left` in case of a successful scan.